doc/specs/#605 - Search/spec.md

```diff
@@ -0,0 +1,125 @@
+8. If the user clicks on the terminal when the search box is open, it will draw focus back to the terminal from the search box. The search box will still stay open. 
+9. The user can interact with the terminal when the search box is open, which means that ther user can scroll the terminal content, or input text. 
+10. If the user switches tabs while the search box is open, the re-focusing will move focus to the terminal. 
+
+## Capabilities
+
+1. The user can search exact matched text in the text buffer of the Terminal Screen. 
+2. The user can choose to search case sensitively and insensitively. 
+3. The user can search up or down. 
+4. Found text will be selected. 
+5. The search will start from the active selected text (inclusive) if there is one, or where the cursor is. 
+5. The search will automatically go around when it reaches the starting point.
+7. The search box can be positioned at either the top left or the bottom left of the terminal. 
+8. The user can use Tab to navigate all the elements in the search box. 
+
+### Accessibility
+
+The user should be able to use search by keyboard only.
+Once the searchbox is focused, the user can navigate between elements in the search box using Tab. And "click" using Enter. 
+
+### Security
+
+This feature should not introduce any new security issues.
+
+### Reliability
+
+1. The key input of Terminal command line and the search box should be separated. search box should not block interaction with the command line when it is open. 
+2. The search box should not block text. This issue could be addressed by make search box movable. 
+
+### Compatibility
+
+This feature won't break existing features of Terminal.
+
+### Performance, Power, and Efficiency
+
+This feature only launches in need. It does not impact the performance of Terminal. 
+
+## Potential Issues
+                                                              
+1. If the terminal window is not wide enough for the search box to be visible, the buttons on the right of the `TextBox` will become invisible, but the `TextBox` is still visible and the window could not be narrower than the `TextBox`. This is similar to the behavior of other editors. 
+2. If the terminal window is not high enough for the search box to be visible, the whole terminal screen, inlcuding the `SearchBoxControl` can disappear. This is similar to the behavior of other editors.
+
+## Future considerations
+
+In version 1, we want realize a case sensitive/insensitive exact text match. But we may consider the following features in version 2:
+
+1. Add "Find" button in dropdown menu to trigger search. This enables the search feature to be operated with mouse only. However, this is not required by Accessibility so we do not cover this in phrase one. 

```

Comment:
```suggestion
1. Add "Find" button in dropdown menu to trigger search.
```
The dropdown menu is accessible via the keyboard. So the accessibility concern doesn't exist.

Reply:
Actually, I mean "the user can use mouse to launch the search dialog" here.